src/index.js

```diff
@@ -191,7 +255,7 @@ class Dropzone extends React.Component {
     }
 
     const inputAttributes = {
-      accept,
+      accept: !(supportMultiple && multiple) ? accept : null,
```

Comment:
The `accept` should still be respected but on the file level IMO. Probably it should be called in the walker callback?